docs/primitives/pubsub.md

```diff
@@ -38,14 +38,36 @@ var Signups = pubsub.NewTopic[*SignupEvent]("signups", pubsub.TopicConfig{
 
 ### At-least-once delivery
 
-This topic behavior configuration ensures that for each subscription to a topic, events will be delivered _at least once_.
+This topic behavior configuration ensures that for each subscription to a topic, events will be delivered _at least once_
+per subscription.
 
 This means that if the topic believes the event was not processed, it will attempt to deliver the message again.
 **Therefore, all subscription handlers should be [idempotent](https://en.wikipedia.org/wiki/Idempotence#Computer_science_meaning).** This helps ensure that if the handler is called two or more times, from the outside there's no difference compared to calling it once.
 
 This can be achieved using a database to track if you have already performed the action that the event is meant to trigger,
 or ensuring that the action being performed is also idempotent in nature.
 
+### Exactly-once delivery
+
+This topic behaviour configuration ensures that for each subscription to a topic, events will be delivered
+_exactly once_ per subscription.
+
+However, there are edge cases when a message might be redelivered.  For example, if a networking issue causes
+the acknowledgement of successful processing the message to be lost before the cloud provider receives it. Thus as with 
+At-least-once delivery it is still advisable to ensure that your subscription handlers are idempotent.
```

Comment:
```suggestion
```suggestion
However, there are still some rare circumstances when a message might be redelivered.  For example, if a
networking issue causes the acknowledgement of successful processing the message to be lost before the cloud provider receives it (the [Two Generals' Problem](https://en.wikipedia.org/wiki/Two_Generals%27_Problem)).
As such, if correctness is critical under all circumstances, it's still advisable to design your subscription handlers to be idempotent.
```
```